promql/functions.go

```diff
@@ -32,29 +33,41 @@ type Function struct {
 	ArgTypes   []ValueType
 	Variadic   int
 	ReturnType ValueType
-	Call       func(ev *evaluator, args Expressions) Value
+
+	// vals is a list of the evaluated arguments for the function call.
+	//    For range vectors it will be a Matrix with one series, instant vectors a
+	//    Vector, scalars a Vector with one series whose value is the scalar
+	//    value,and nil for strings.
+	// args are the original arguments to the function, where you can access
+	//    matrixSelectors, vectorSelectors, and StringLiterals.
+	// enh.out is a pre-allocated empty vector that you may use to accumulate
+	//    output before returning it. The vectors in vals should not be returned.a
+	// Range vector functions need only return a vector with the right value,
+	//     the metric and timestamp are not neded.
+	// Instant vector functions need only return a vector with the right values and
+	//     metrics, the timestamp are not needed.
+	// Scalar results should be returned as the value of a sample in a Vector.
+	Call func(vals []Value, args Expressions, enh *evalNodeHelper) Vector
```

Comment:
Via `evalNodeHelper` we now have an exported type referencing an unexported type, i.e. it cannot actually be used outside of the package.